korgw/src/jvmMain/kotlin/com/soywiz/korgw/awt/AwtAg.kt

```diff
@@ -15,12 +17,27 @@ class AwtAg(override val nativeComponent: Any, private val checkGl: Boolean, log
     private var baseLazyGlWithLog: LogKmlGlProxy? = null
     private var lazyGl: KmlGlFastProxy? = null
 
-    override var devicePixelRatio: Double = 1.0
-        get() {
+    companion object {
+        const val defaultPixelsPerLogicalInch = 96.0
+    }
+
+    override val devicePixelRatio: Double by lazy(LazyThreadSafetyMode.PUBLICATION)
+        {
+            if (GraphicsEnvironment.isHeadless()) {
+                return@lazy 1.0
+            }
+            // transform
             // https://stackoverflow.com/questions/20767708/how-do-you-detect-a-retina-display-in-java
             val config = (nativeComponent as? Component)?.graphicsConfiguration
                 ?: GraphicsEnvironment.getLocalGraphicsEnvironment().defaultScreenDevice.defaultConfiguration
-            return config.defaultTransform.scaleX
+            val scale = config.defaultTransform.scaleX
+
+            // maybe this is not just windows specific :
+            // https://stackoverflow.com/questions/32586883/windows-scaling
+            // somehow this value is not update when you change the scaling in the windows settings while the jvm is running :(
+            val pixelsPerLogicalInchRatio = Toolkit.getDefaultToolkit().screenResolution / defaultPixelsPerLogicalInch
+
+            return@lazy (scale * pixelsPerLogicalInchRatio)
```

Comment:
Tested on mac. On mac, it should be something like `scale / pixelsPerLogicalInchRatio`.

But I believe we should separate those values:

```
override val devicePixelRatio: Double get() = ... scaleX
override val pixelsPerLogicalInchRatio: Double get() = ... screenResolution / defaultPixelsPerLogicalInch
// Use this in the debug handler, while allowing people to access raw devicePixelRatio without the noise of window scaling
override val computedPixelRatio: Double get() = devicePixelRatio / pixelsPerLogicalInchRatio
```

Otherwise `devicePixelRatio` stops being true. Is it 1 on your hdpi screen? On mac devices with retina displays it is 2, without it, it is 1.

Reply:
But yes we can separate the values.

Reply:
i'm still not sure about the calculation of "computedPixelRatio" becasue in my opinion it must be 

`devicePixelRatio * pixelsPerLogicalInchRatio` 
or
`pixelsPerLogicalInchRatio / devicePixelRatio `